### ISSUE TITLE
[flagd-provider] Can't import flagd provider in ESM Typescript project

### ISSUE THREAD

**Issue by talzion12** [open-feature/js-sdk-contrib#198]
Trying to import FlagdProvider in Typescript with ESM causes the following error:
```
node_modules/@openfeature/flagd-provider/src/index.d.ts:1:15 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

1 export * from './lib/flagd-provider';
                ~~~~~~~~~~~~~~~~~~~~~~

src/main.ts:2:10 - error TS2305: Module '"@openfeature/flagd-provider"' has no exported member 'FlagdProvider'.

2 import { FlagdProvider } from '@openfeature/flagd-provider';
           ~~~~~~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  node_modules/@openfeature/flagd-provider/src/index.d.ts:1
     1  src/main.ts:2
```

When I switch my project to CommonJS the problem is resolved.
I'm not sure if this is a bug or if I'm doing something wrong here.

### Steps to reproduce

* Create Typescript ESM project.
* Import FlagdProvider like so: `import { FlagdProvider } from '@openfeature/flagd-provider';`.

I also created a simple [example repo](https://github.com/talzion12/openfeature-example). Running `npm run build` shows the error.

**Comment by beeme1mr** [comment]
Hi @talzion12, thanks for reporting this. Sorry you keep running into ESM issues! The team has limited availability this week but we'll address it as soon as possible.

**Comment by talzion12** [comment]
@beeme1mr Haha yes ESM is causing me quite a bit of issues 😅
No problem, in the meantime I created a custom provider.
Thanks!

**Comment by toddbaert** [comment]
I'll look into this today and hopefully publish a fix.

**Comment by toddbaert** [comment]
Thanks again for the report @talzion12 . Looks like this might have been a [change/regression in NX](https://github.com/open-feature/js-sdk-contrib/pull/201).

I published `0.7.3` which resolves the issue in your example project.

**Comment by toddbaert** [comment]
I'm still seeing issues with this. I have a fix I'll push tomorrow.
